### PR TITLE
Install fixes

### LINKED ISSUE #97: When starting the bot for the first time, a database error is displayed in the console
```
[ERR]   :  Select in database failed : SELECT value FROM _Migrations WHERE name = $name
```
While this is expected because migrations haven't been done yet, it would be nice to not display it if we just created the database.

### LINKED ISSUE #98: When the bot is started for the first time, owner insertion fails
```
[2/22/2020 6:32:25 PM] [ERR]   :  Insert in database failed : INSERT INTO Owners (discordUserId) VALUES ($discordUserId)
[2/22/2020 6:32:25 PM] [ERR]   :  { [Error: SQLITE_CONSTRAINT: FOREIGN KEY constraint failed] errno: 19, code: 'SQLITE_CONSTRAINT' }
```
This happens the first the bot is ran, restarting it "fixes" the issue

### PR DESCRIPTION
**Pull request by Blatoy**

### CODE REVIEW

File: bot/utils/database-migrations.js [hunk 1]
```diff
@@ -26,25 +26,21 @@ module.exports.latestMigration = () => {
  * Gets the currently applied migration.
  * @returns {Promise<string>} Promise resolving to the current migration's full name.
  */
-module.exports.current = () => {
-    return db.select("_Migrations", ["value"], {
-        name: "version"
-    }).then(results => {
-        if (results.length === 0) {
+module.exports.current = async () => {
+    const tableCountResult = await db.select("sqlite_master", ["count(*)"]);
```
Comment (RedMser) [inline bot/utils/database-migrations.js:29]: Doesn't this need some form of error handling? What if the database couldn't connect or something like that?

(As a caller, I wouldn't expect a "get current migration" method to raise errors)

File: install.js [hunk 2]
```diff
@@ -123,14 +123,14 @@ if (redirectURL.startsWith("https")) {
...
-  "certificate": "${certificate ? privateKey : ""}"
-}`;
+let data = JSON.stringify({
+    token: token,
+    applicationId: appId,
+    oauth2Secret: appSecret,
+    redirectURI: domain,
+    privateKey: privateKey,
+    certificate: certificate
+});
```
Comment (RedMser) [inline install.js]: I would personally add `, null, 4` to the arguments of stringify, so it adds indentation and newlines.
Makes the config easier to read and edit.

File: bot/index.js [hunk 3]
```diff
@@ -44,19 +44,19 @@ async function init() {
...
     registerEvents();
-    connectToDiscord();
+    await connectToDiscord();
 
     discotron.registerActions();
 
     /**
      * Attempts to connect the bot client to Discord
-     * @returns {boolean} true if login is successful, false otherwise
+     * @returns {Promise} true if login is successful, false otherwise
```
Comment (RedMser) [inline bot/index.js]: Specify type `Promise<boolean>`

File: bot/utils/database-migrations.js (hunk 1 above)
Comment (Blatoy) [reply bot/utils/database-migrations.js:29]: If we can't connect to the db, wouldn't we have errors before this line anyway?

Comment (RedMser) [reply bot/utils/database-migrations.js:29]: :thinking: SQL is scary... What if the `_Migrations` table got swallowed by a bit flip?? *Think of the possibilities!*

**Review by RedMser — APPROVED** [review]

**Pull request merged**